fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/communication/CellModule.java

```diff
@@ -119,7 +118,10 @@ public Integer sendCam() {
         if (camConfiguration.getAddressingMode().equals(DestinationType.CELL_TOPOCAST)) {
```

Comment:
Should this maybe be a `switch`-statement for better readability?

Reply:
Yes, that would be more readable, but also much longer:

```java
    public Integer sendCam() {
        if (!isEnabled()) {
            log.warn("sendCAM: Cell communication disabled (!cellModule.isEnabled()).");
            return null;
        }
        if (configuration == null || configuration.getCamConfiguration() == null) {
            log.warn("sendCAM: No camConfiguration with addressingMode and geoRadius given.");
            return null;
        }
        CellModuleConfiguration.CellCamConfiguration camConfiguration = configuration.getCamConfiguration();
        switch (camConfiguration.getAddressingMode()) {
            case CELL_TOPOCAST:
                return sendCamViaTopocast(camConfiguration);
            case CELL_GEOCAST:
                return sendCamViaGeoBroadcast(camConfiguration);
            case CELL_GEOCAST_MBMS:
                return sendCamViaGeoBroadcastMbms(camConfiguration);
            default:
                log.warn("sendCam: Unsupported addressing mode {}.", camConfiguration.getAddressingMode());
                return null;
        }
    }

    private Integer sendCamViaTopocast(CellModuleConfiguration.CellCamConfiguration camConfiguration) {
        return super.sendCam(createMessageRouting().topoCast(camConfiguration.getTopocastReceiver()));
    }

    private Integer sendCamViaGeoBroadcast(CellModuleConfiguration.CellCamConfiguration camConfiguration) {
        if (!(getOwner() instanceof Locatable)) {
            throw new UnsupportedOperationException("Cannot send CAM for entities without a location.");
        }
        final GeoCircle destination = new GeoCircle(((Locatable) getOwner()).getPosition(), camConfiguration.getGeoRadius());
        return super.sendCam(createMessageRouting().geoBroadcastBasedOnUnicast(destination));
    }

    private Integer sendCamViaGeoBroadcastMbms(CellModuleConfiguration.CellCamConfiguration camConfiguration) {
        if (!(getOwner() instanceof Locatable)) {
            throw new UnsupportedOperationException("Cannot send CAM for entities without a location.");
        }
        final GeoCircle destination = new GeoCircle(((Locatable) getOwner()).getPosition(), camConfiguration.getGeoRadius());
        return super.sendCam(createMessageRouting().geoBroadcastMbms(destination));
    }
```